src/model2d.py

```diff
@@ -0,0 +1,158 @@
+import math
+import numpy as np
+import unittest
+import pdb
+
+class DronePos:
+	""" Describes the position and rotation of the drone. """
+	def __init__(self, x, y, t):
+		self.x = x
+		self.y = y
+		self.t = t
+
+	def __str__(self):
+		return "({0}, {1}, {2})".format(self.x, self.y, self.t)
+
+	def __repr__(self):
+		return self.__str__()
+
+class DroneVel:
+	""" Describes the velocity and angular velocity of the drone. """
+	def __init__(self, vx, vy, w):
+		self.vx = vx
+		self.vy = vy
+		self.w = w
+
+class Drone2d:
+	""" The model of a 2d bicopter """
+	cMaxThrust = 20
+	cRadius = 0.5
+
+	def __init__(self):
+		""" By default, the drone is 1kg, positioned at 0,0 with no
+		rotation, and traveling at 0 velocity and angular velocity. """
+		self.initialize(DronePos(0, 0, 0), DroneVel(0, 0, 0), 1)
+		self.thrustL = 0
+		self.thrustR = 0
+
+	def initialize(self, initialPos, initialVel, mass):
+		self.pos = initialPos
+		self.vel = initialVel
+		self.mass = mass
+
+	def setThrustL(self, thrust):
+		thrust = max(thrust, 0)
+		self.thrustL = min(thrust, self.cMaxThrust)
+
+	def setThrustR(self, thrust):
+		thrust = max(thrust, 0)
+		self.thrustR = min(thrust, self.cMaxThrust)
+
+	def getThrust(self):
+		totalThrust = self.thrustL + self.thrustR
+		# Thrust is along the Y axis when the angle is 0.
+		thrustX = totalThrust * math.sin(self.pos.t)
+		thrustY = totalThrust * math.cos(self.pos.t)
+		return [thrustX, thrustY]
+
+	def getAngularAccel(self):
+		""" Angular acceleration due to thrust, assuming that the
+		drone has uniform mass and is a rod of length 2 * cRadius """
+		thrustDiff = self.thrustL - self.thrustR
```

Comment:
Probably should document that thrust is assumed to be perpendicular to the body and therefore the torque is equal in magnitude to the tangential force.